[CI] MixedClusterEsqlSpecIT test {date.EvalDateFormatString SYNC} failing

**Build Scans:**
- [elasticsearch-intake #9960 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/aw7t7fqb2n73s)
- [elasticsearch-pull-request #32281 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/rpssfmkfwcxnm)
- [elasticsearch-pull-request #32273 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/fercqmyv5dzhc)
- [elasticsearch-pull-request #32264 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/gofowgcf2q5q2)
- [elasticsearch-intake #9957 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/veahagtflgb3o)
- [elasticsearch-pull-request #32265 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/bi6jevozqt4gu)
- [elasticsearch-pull-request #32263 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/prxlfmvyvu2yu)
- [elasticsearch-pull-request #32183 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/n2xq24mo64g6y)
- [elasticsearch-pull-request #32260 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/4sayh2tkq4lye)
- [elasticsearch-pull-request #32258 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/qnlqmv6l5sq62)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.16.0#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {date.EvalDateFormatString SYNC}" -Dtests.seed=11953366FC0CD373 -Dtests.bwc=true -Dtests.locale=twq-NE -Dtests.timezone=America/Cancun -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bdate.EvalDateFormatString%20SYNC%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Expected an empty list
0: <unexpected> but was "Date format [YYYY-MM-dd] contains week-date field specifiers that are changing in JDK 23"
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {date.EvalDateFormatString SYNC}
- [main] 2 consecutive failures in step 8.16.0_bwc-snapshots
- [main] 13 failures in test test {date.EvalDateFormatString SYNC} (1.4% fail rate in 944 executions)
- [main] 11 failures in step 8.16.0_bwc-snapshots (2.6% fail rate in 427 executions)
- [main] 2 failures in pipeline elasticsearch-intake (1.6% fail rate in 125 executions)
- [main] 10 failures in pipeline elasticsearch-pull-request (2.8% fail rate in 362 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -265,20 +265,6 @@ tests:
 - class: org.elasticsearch.xpack.security.authz.SecurityScrollTests
   method: testSearchAndClearScroll
   issue: https://github.com/elastic/elasticsearch/issues/113285
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {date.EvalDateFormatString SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/113293
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {date.EvalDateFormat}
-  issue: https://github.com/elastic/elasticsearch/issues/113294
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {date.DocsDateFormat}
-  issue: https://github.com/elastic/elasticsearch/issues/113295
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {stats.DocsStatsGroupByMultipleValues}
-  issue: https://github.com/elastic/elasticsearch/issues/113296
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  issue: https://github.com/elastic/elasticsearch/issues/113298
 - class: org.elasticsearch.integration.KibanaUserRoleIntegTests
   method: testGetIndex
   issue: https://github.com/elastic/elasticsearch/issues/113311

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
@@ -46,7 +46,7 @@ emp_no:integer | x:date
 
 evalDateFormat
 from employees | sort hire_date | eval x = date_format(hire_date), y = date_format("YYYY-MM-dd", hire_date) | keep emp_no, x, y | limit 5;
-warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
+warningRegex:Date format \[YYYY\-MM\-dd\] contains week-date field specifiers that are changing in JDK 23
 
 emp_no:integer | x:keyword                     | y:keyword
 10009          | 1985-02-18T00:00:00.000Z      | 1985-02-18            
@@ -962,7 +962,7 @@ FROM employees
 | SORT first_name
 | LIMIT 3
 ;
-warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
+warningRegex:Date format \[YYYY\-MM\-dd\] contains week-date field specifiers that are changing in JDK 23
 
 // tag::docsDateFormat-result[]
 first_name:keyword   |   last_name:keyword   | hire_date:date           | hired:keyword
@@ -978,7 +978,7 @@ required_capability: string_literal_auto_casting
 ROW a = 1
 | EVAL df = DATE_FORMAT("YYYY-MM-dd", "1989-06-02T00:00:00.000Z")
 ;
-warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
+warningRegex:Date format \[YYYY\-MM\-dd\] contains week-date field specifiers that are changing in JDK 23
 
 a:integer | df:keyword
 1         | 1989-06-02

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/date.csv-spec
@@ -46,6 +46,7 @@ emp_no:integer | x:date
 
 evalDateFormat
 from employees | sort hire_date | eval x = date_format(hire_date), y = date_format("YYYY-MM-dd", hire_date) | keep emp_no, x, y | limit 5;
+warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
 
 emp_no:integer | x:keyword                     | y:keyword
 10009          | 1985-02-18T00:00:00.000Z      | 1985-02-18            
@@ -961,6 +962,7 @@ FROM employees
 | SORT first_name
 | LIMIT 3
 ;
+warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
 
 // tag::docsDateFormat-result[]
 first_name:keyword   |   last_name:keyword   | hire_date:date           | hired:keyword
@@ -976,6 +978,7 @@ required_capability: string_literal_auto_casting
 ROW a = 1
 | EVAL df = DATE_FORMAT("YYYY-MM-dd", "1989-06-02T00:00:00.000Z")
 ;
+warningRegex:Date format \\[YYYY.MM.dd\\] contains week-date field specifiers that are changing in JDK 23
 
 a:integer | df:keyword
 1         | 1989-06-02

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
@@ -1183,6 +1183,7 @@ FROM employees
 // end::statsGroupByMultipleValues[]
 | LIMIT 4
 ;
+warningRegex:Date format \\[YYYY\\] contains week-date field specifiers that are changing in JDK 23
 
 hired:keyword |languages.long:long | avg_salary:double
 1985           |1              |54668.0        

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
@@ -1183,7 +1183,7 @@ FROM employees
 // end::statsGroupByMultipleValues[]
 | LIMIT 4
 ;
-warningRegex:Date format \\[YYYY\\] contains week-date field specifiers that are changing in JDK 23
+warningRegex:Date format \[YYYY\] contains week-date field specifiers that are changing in JDK 23
 
 hired:keyword |languages.long:long | avg_salary:double
 1985           |1              |54668.0        

